### PR TITLE
[BEAM-11205] Update Libraries BOM dependencies to version 26.1.1

### PR DESCRIPTION
**Pull request by rajatbhatta**
Updating Google Cloud Libraries BOM to the latest version 26.1.1 and associated dependencies to the compatible versions.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by rajatbhatta** [comment]
R: @suztomo

**Comment by kileys** [comment]
Did you get a chance to run the linkage checker? https://github.com/apache/beam/blob/master/sdks/java/build-tools/beam-linkage-check.sh

**Comment by pabloem** [comment]
Run Kotlin_Examples PreCommit

**Comment by pabloem** [comment]
Run Java_Examples_Dataflow_Java17 PreCommit

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by kileys** [comment]
Run Java PostCommit

**Comment by kileys** [comment]
Run Dataflow ValidatesRunner

**Comment by kileys** [comment]
Run PostCommit_Java_Dataflow

**Comment by kileys** [comment]
Run PostCommit_Java_DataflowV2

**Comment by kileys** [comment]
Run SQL Postcommit

**Comment by kileys** [comment]
Run Spark ValidatesRunner

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -464,15 +464,15 @@ class BeamModulePlugin implements Plugin<Project> {
     def classgraph_version = "4.8.104"
     def errorprone_version = "2.10.0"
     // Try to keep gax_version consistent with gax-grpc version in google_cloud_platform_libraries_bom
-    def gax_version = "2.16.0"
+    def gax_version = "2.18.7"
     def google_clients_version = "1.32.1"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:468]: Can you upgrade this `google_clients_version` to 2.0.0? As per  https://github.com/googleapis/java-shared-dependencies/blob/v3.0.1/first-party-dependencies/pom.xml#L69. This is for `com.google.api-client:google-api-client` in line 574. ([Maven link](https://search.maven.org/artifact/com.google.api-client/google-api-client/2.0.0/jar)).

Comment (suztomo) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:468]: Then, update line 578 - 584 as followings:

```
        google_api_services_bigquery                : "com.google.apis:google-api-services-bigquery:v2-rev20220827-$google_clients_version",
        google_api_services_clouddebugger           : "com.google.apis:google-api-services-clouddebugger:v2-rev20220318-$google_clients_version",
        google_api_services_cloudresourcemanager    : "com.google.apis:google-api-services-cloudresourcemanager:v3-rev20220828-$google_clients_version",
        google_api_services_dataflow                : "com.google.apis:google-api-services-dataflow:v1b3-rev20220812-$google_clients_version",
        google_api_services_healthcare              : "com.google.apis:google-api-services-healthcare:v1-rev20220818-$google_clients_version",
        google_api_services_pubsub                  : "com.google.apis:google-api-services-pubsub:v1-rev20220829-$google_clients_version",
        google_api_services_storage                 : "com.google.apis:google-api-services-storage:v1-rev20220705-$google_clients_version",
```

The "google-api-services-*" libraries need to have "-2.0.0" suffix version to work with google-api-client version 2.0.0. (Details https://github.com/googleapis/google-api-java-client-services/issues/13338 )

Comment (rajatbhatta) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Incorporated the changes.

Comment (rajatbhatta) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: There are a lot of failed github checks after this. Can you please check this?

Comment (suztomo) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: What is error messages?

Comment (suztomo) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]:
```
/home/runner/work/beam/beam/sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptions.java:27: error: package com.google.api.services.cloudresourcemanager does not exist
import com.google.api.services.cloudresourcemanager.CloudResourceManager;
                                                   ^
/home/runner/work/beam/beam/sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptions.java:28: error: package com.google.api.services.cloudresourcemanager.model does not exist
import com.google.api.services.cloudresourcemanager.model.Project;
```

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -575,13 +575,13 @@ class BeamModulePlugin implements Plugin<Project> {
...
-        google_api_services_cloudresourcemanager    : "com.google.apis:google-api-services-cloudresourcemanager:v1-rev20211017-$google_clients_version",
-        google_api_services_dataflow                : "com.google.apis:google-api-services-dataflow:v1b3-rev20210818-$google_clients_version",
-        google_api_services_healthcare              : "com.google.apis:google-api-services-healthcare:v1-rev20211016-$google_clients_version",
-        google_api_services_pubsub                  : "com.google.apis:google-api-services-pubsub:v1-rev20211130-$google_clients_version",
-        google_api_services_storage                 : "com.google.apis:google-api-services-storage:v1-rev20211201-$google_clients_version",
+        google_api_services_bigquery                : "com.google.apis:google-api-services-bigquery:v2-rev20220827-$google_clients_version",
+        google_api_services_clouddebugger           : "com.google.apis:google-api-services-clouddebugger:v2-rev20220318-$google_clients_version",
+        google_api_services_cloudresourcemanager    : "com.google.apis:google-api-services-cloudresourcemanager:v3-rev20220828-$google_clients_version",
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:580]: Use `v1-rev20220828-$google_clients_version` for google-api-services-cloudresourcemanager.

The previous version `v1-rev20211017-$google_clients_version` started with "v1". It seems it should still continue to use "v1-" version.

Comment (rajatbhatta) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Thanks Tomo. Have pushed the fix.

**Comment by suztomo** [comment]
Java PreCommit failed https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/23860/consoleFull :

```
09:43:59 > Task :sdks:java:testing:test-utils:compileJava
09:43:59 warning: Memory constraints are impeding performance; please increase max heap size.
09:44:02 error: warnings found and -Werror specified
09:44:07 
...
09:44:10 > Task :sdks:java:testing:test-utils:compileJava FAILED
09:44:10 1 error
09:44:10 1 warning

10:22:59 * What went wrong:
10:22:59 Execution failed for task ':sdks:java:testing:test-utils:compileJava'.
10:22:59 > Compilation failed; see the compiler error output for details.
```

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run PostCommit_Java_Hadoop_Versions

**Comment by suztomo** [comment]
Run PostCommit_Java_Dataflow

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by rajatbhatta** [comment]
Run PostCommit_Java_Hadoop_Versions

**Comment by suztomo** [comment]
### grpc-google-cloud-bigtable-v2 version

```
(com.google.api.grpc:grpc-google-cloud-bigtable-v2:1.27.1) com.google.bigtable.v2.BigtableGrpc's method getPingAndWarmMethod() is not found;
  referenced by 1 class file
    com.google.cloud.bigtable.data.v2.stub.EnhancedBigtableStub (com.google.cloud:google-cloud-bigtable:2.11.0)
  Cause:
    Dependency conflict: com.google.api.grpc:grpc-google-cloud-bigtable-v2:1.27.1 does not define com.google.bigtable.v2.BigtableGrpc's method getPingAndWarmMethod() but com.google.api.grpc:grpc-google-cloud-bigtable-v2:2.11.0 defines it.
      selected: org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.42.0-SNAPSHOT / com.google.cloud.bigtable:bigtable-client-core:1.26.3 (compile) / com.google.api.grpc:grpc-google-cloud-bigtable-v2:1.27.1 (compile)
      unselected: org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.42.0-SNAPSHOT / com.google.cloud:google-cloud-bigtable:2.11.0 (compile) / com.google.api.grpc:grpc-google-cloud-bigtable-v2:2.11.0 (compile)
```

https://gist.github.com/suztomo/9aef72d020f96446b70f4eab8cba948b

On bigtable-client-core, the latest 1.27.1 https://search.maven.org/artifact/com.google.cloud.bigtable/bigtable-client-core/1.27.1/jar declares dependencyManagement to import https://search.maven.org/artifact/com.google.cloud/google-cloud-bigtable-bom/1.27.1/pom (via `bigtable.version` 1.27.1 set in https://search.maven.org/artifact/com.google.cloud.bigtable/bigtable-client-parent/1.27.1/pom), which has grpc-google-cloud-bigtable-v2 version "1.27.1", not 2.11.0.

@kolea2 (Probably not in this pull request) Would you provide guidance what Bigtable-related artifact versions Apache Beam should depend on?

**Comment by suztomo** [comment]
Run Go PreCommit

**Comment by suztomo** [comment]
Run Kotlin_Examples PreCommit

**Comment by kileys** [comment]
Run Python PreCommit

**Comment by kileys** [comment]
Java_GCP_IO_Direct may be a valid failure. Seeing java.lang.NoClassDefFoundError: io/netty/internal/tcnative/CertificateCompressionAlgo

**Comment by kileys** [comment]
io.netty:netty-tcnative-boringssl-static:2.0.47.Final Needs to be updated for the missing class def

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 3]
```diff
@@ -677,7 +677,7 @@ class BeamModulePlugin implements Plugin<Project> {
         nemo_compiler_frontend_beam                 : "org.apache.nemo:nemo-compiler-frontend-beam:$nemo_version",
         netty_all                                   : "io.netty:netty-all:$netty_version",
         netty_handler                               : "io.netty:netty-handler:$netty_version",
-        netty_tcnative_boringssl_static             : "io.netty:netty-tcnative-boringssl-static:2.0.47.Final",
+        netty_tcnative_boringssl_static             : "io.netty:netty-tcnative-boringssl-static:2.0.52.Final",
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:680]: Netty_version is now 4.1.77.Final. It depends on `<tcnative.version>2.0.52.Final</tcnative.version>`:
https://search.maven.org/artifact/io.netty/netty-parent/4.1.77.Final/jar

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run PostCommit_Java_DataflowV2

**Comment by suztomo** [comment]
Run PostCommit_Java_Dataflow

**Comment by suztomo** [comment]
Run Spotless PreCommit

**Comment by suztomo** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
52 successful and 1 skipped checks

**Review by suztomo — APPROVED** [review]

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 4]
```diff
@@ -575,13 +575,13 @@ class BeamModulePlugin implements Plugin<Project> {
...
-        google_api_services_storage                 : "com.google.apis:google-api-services-storage:v1-rev20211201-$google_clients_version",
+        google_api_services_bigquery                : "com.google.apis:google-api-services-bigquery:v2-rev20220827-$google_clients_version",
+        google_api_services_clouddebugger           : "com.google.apis:google-api-services-clouddebugger:v2-rev20220318-$google_clients_version",
+        google_api_services_cloudresourcemanager    : "com.google.apis:google-api-services-cloudresourcemanager:v1-rev20220828-$google_clients_version",
+        google_api_services_dataflow                : "com.google.apis:google-api-services-dataflow:v1b3-rev20220812-$google_clients_version",
+        google_api_services_healthcare              : "com.google.apis:google-api-services-healthcare:v1-rev20220818-$google_clients_version",
+        google_api_services_pubsub                  : "com.google.apis:google-api-services-pubsub:v1-rev20220829-$google_clients_version",
+        google_api_services_storage                 : "com.google.apis:google-api-services-storage:v1-rev20220705-$google_clients_version",
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:584]: With `def google_clients_version = "2.0.0"` (line 468), it should avoid https://github.com/googleapis/google-api-java-client-services/issues/13338.

**Pull request merged**